clusterdock/topologies/cdh/actions.py

```diff
@@ -162,18 +198,48 @@ def start(args):
                 deployment.cluster.delete_service(service.name)
 
     hue_server_host_port = get_host_port_binding(primary_node.container_id, HUE_SERVER_PORT)
+    is_hue_installed = False
     for service in deployment.cluster.get_all_services():
         if service.type == 'HUE':
             logger.info("Once its service starts, Hue server will be accessible at http://%s:%s",
                         getfqdn(), hue_server_host_port)
+            is_hue_installed = True
             break
 
-    logger.info("Deploying client configuration...")
-    deployment.cluster.deploy_client_config().wait()
+    if args.kerberos:
+        logger.info('Updating config for kerberos...')
+        kerberos_config = {'SECURITY_REALM': args.network.upper(),
+                           'KDC_HOST': KDC_HOST_NAME,
+                           'KRB_MANAGE_KRB5_CONF': True,
+                           'KRB_ENC_TYPES': 'aes256-cts-hmac-sha1-96'}
+        deployment.cm.update_config(kerberos_config)
+
+        logger.info('Importing kerberos admin credentials...')
+        deployment.cm.import_admin_credentials('{}@{}'.format(CLDR_MGR_PRINCIPAL_USER, args.network.upper()),
+                                               CLDR_MGR_PRINCIPAL_PW).wait(60)
+
+        logger.info('Configuring for kerberos...')
+        config_kerberos_cmd = deployment.cluster.configure_for_kerberos()
+        if not config_kerberos_cmd.wait(360).success:
+            raise Exception('Failed to configure for kerberos.')
+
+        logger.info('Deploying cluster client configuration...')
+        cluster_client_config_cmd = deployment.cluster.deploy_cluster_client_config()
+        if not cluster_client_config_cmd.wait(360).success:
+            raise Exception('Failed to deploy cluster client configurations.')
+
+        if is_hue_installed:
+            hue_node_name = '{}.{}'.format(args.primary_node[0], args.network)
+            _fix_for_hue_kerberos(cluster, [kdc_node], args.network.upper(), hue_node_name)
+
+    logger.info('Deploying client configuration...')
+    client_config_cmd = deployment.cluster.deploy_client_config()
+    if not client_config_cmd.wait(360).success:
+        raise Exception("Failed to deploy client configurations.")
 
     if not args.dont_start_cluster:
         logger.info('Starting cluster...')
-        if not deployment.cluster.start().wait().success:
+        if not deployment.cluster.start().wait(720).success:
```

Comment:
Why `720`? Perhaps make a constant at the top of the file instead? I've also seen `60` and `360` used. Any reason to pick a shorter timeout value if a longer one will suffice?